handlers/handlers.go

```diff
@@ -150,62 +119,197 @@ func (d *CatalystAPIHandlersCollection) processUploadVOD(streamName, sourceURL,
 	return nil
 }
 
-func randomStreamName(prefix string) string {
+func randomTrailer() string {
 	const charset = "abcdefghijklmnopqrstuvwxyz0123456789"
 	const length = 8
 	r := rand.New(rand.NewSource(time.Now().UnixNano()))
-
 	res := make([]byte, length)
 	for i := 0; i < length; i++ {
 		res[i] = charset[r.Intn(length)]
 	}
-	return fmt.Sprintf("%s%s", prefix, string(res))
+	return string(res)
+}
+func randomStreamName(prefix string) string {
+	return fmt.Sprintf("%s%s", prefix, randomTrailer())
 }
 
 type MistCallbackHandlersCollection struct {
 	MistClient  MistAPIClient
-	StreamCache map[string]StreamInfo
+	StreamCache *StreamCache
 }
 
-func (d *MistCallbackHandlersCollection) Trigger() httprouter.Handle {
-	return func(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
-		if t := req.Header.Get("X-Trigger"); t != "PUSH_END" {
-			errors.WriteHTTPBadRequest(w, "Unsupported X-Trigger", fmt.Errorf("unknown trigger '%s'", t))
-			return
+// This trigger is stream-specific and must be blocking.
```

Comment:
Please change all comments to follow Golang standards. In this case, it should start from the function name.